File: src/server/Hoard.Core/Application/Transactions/TransactionSummaryDto.cs
namespace Hoard.Core.Application.Transactions;

public class TransactionSummaryDto
{
    public int Id { get; set; }
    
    public int AccountId { get; set; }
    public required string AccountName { get; set; }
    
    public int? InstrumentId { get; set; }
    public required string InstrumentName { get; set; }
    public required string InstrumentTicker { get; set; }
    
    public int TransactionTypeId { get; set; }
    public required string TransactionTypeName { get; set; }
    
    public DateOnly Date { get; set; }
    public required string ContractNoteReference { get; set; }
    
    public decimal? Price { get; set; }
    public decimal? Units { get; set; }
    public decimal Value { get; set; }
}

File: src/client/hoard-ui/src/api/dtos/Transactions/TransactionSummaryDto.ts
export interface TransactionSummaryDto {
  id: number
  accountId: number
  accountName: string
  contractNoteReference: string
  date: string // ISO date: YYYY-MM-DD
  instrumentId?: number
  instrumentTicker?: string
  instrumentName?: string
  transactionTypeId: number
  transactionTypeName: string
  units?: number
  value: number
  price?: number
}
